Update primary buttons on signup

adamwoodnz opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to https://linear.app/a8c/issue/DSGCOM-93/update-primary-buttons-on-signup

## Proposed Changes

Update the signup forms:

- Replace the signup A8C primary button with a core button to fix accessibility issues and bring consistency with login buttons (replaced in https://github.com/Automattic/wp-calypso/pull/102639) and social buttons (replaced in https://github.com/Automattic/wp-calypso/pull/100807)
- Standardize the height (40px) and border radius (2px) of text inputs and buttons, unless the brand has specific styling (like Woo)
- Add stories for each product to help in checking for regressions, eg. when upgrading core components

## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

These signup screens have inconsistent buttons heights and border radii, missing or incorrectly applied brand colors, and some have no visible focus styles for the primary button.

- Focus styles are necessary for accessibility
- Consistent button styles and behaviours form a more high quality experience

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

### Storybook
1. Run `yarn storybook: start` and the storybook should open in your browser.
2. Browse the sign up submit button stories at http://localhost:6006/?path=/story/client-blocks-signup-submit-button--default
3. Test using prop toggles, hover and focus states, and brands

![Screenshot 2025-05-08 at 2 24 28 PM](https://github.com/user-attachments/assets/17b0f2ec-6b2a-4508-b17f-514503338fa4)

### Product

1. Open the signup screen for each product (see links below)
2. Check the interactive states for each primary button (particularly hover and focus) - hover background should generally use a darker product accent color (some don't have this, eg. Akismet, Blaze), and the focus ring color should use the accent color. Compare with other buttons, eg social. Focus ring should be the same, but background change should be different.
3. Tab through the form to ensure the buttons are focusable
4. Activate the buttons to ensure the actions still fire
5. Check for responsiveness - mobile and ipad size (Crowdsignal in particular has quite a different responsive layout)

#### Signup links

[WordPress.com](http://calypso.localhost:3000/start/account/user-social) -> 'Continue with email' button
[Akismet](http://calypso.localhost:3000/start/account/user-social?redirect_to=https%3A%2F%2Fr-login.wordpress.com%2Fremote-login.php%3Faction%3Dlink%26back%3Dhttps%253A%252F%252Fakismet.com%252Faccount%252F) -> 'Continue with email' button
[Woo](http://calypso.localhost:3000/start/wpcc/oauth2-user?oauth2_client_id=50916&oauth2_redirect=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%2F%3Fresponse_type%3Dcode%26client_id%3D50916%26state%3D36e78dc55c52c1680dc1b15a5e019bddb1baab60ee71bbedf5004689c43f491b%26redirect_uri%3Dhttps%253A%252F%252Fwoocommerce.com%252Fwc-api%252Fwpcom-signin%253Fnext%253D%25252F%26blog_id%3D0%26wpcom_connect%3D1%26wccom-from%26calypso_env%3Dproduction%26from-calypso%3D1)
[Blaze Pro](http://calypso.localhost:3000/start/wpcc/oauth2-user?oauth2_client_id=99370&oauth2_redirect=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fresponse_type%3Dcode%26client_id%3D99370%26state%3D6e545c7a9932a0a612817faa07992930afae8229e4b11a219e9db95944038cb9%26redirect_uri%3Dhttps%253A%252F%252Fblazepro.tumblr.com%252Fauth%252Fconnected%26blog_id%3D0%26wpcom_connect%3D1%26calypso_env%3Dproduction%26scope%3Dglobal%26from-calypso%3D1)
[Jetpack Cloud](http://calypso.localhost:3000/start/wpcc/oauth2-user?oauth2_client_id=69040&oauth2_redirect=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fresponse_type%3Dtoken%26client_id%3D69040%26redirect_uri%3Dhttps%253A%252F%252Fcloud.jetpack.com%252Fconnect%252Foauth%252Ftoken%253Fnext%253D%25252F%26scope%3Dglobal%26blog_id%3D0%26from-calypso%3D1)
A4A - Can't link due to GitHub OSS scan rules: go to automatt_c.com/for-agencies/ and click 'Log in', then 'Create an Account'. Edit the URL and change the host to calypso.localhost:3000
[VIP](http://calypso.localhost:3000/start/wpcc/oauth2-user?oauth2_client_id=76596&oauth2_redirect=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fstate%3D2e373ba7d0b99c5b66af232a2cd4a90f3af2aeafbb0326604917c9333ac1%26scope%3Dauth%26response_type%3Dcode%26approval_prompt%3Dauto%26redirect_uri%3Dhttps%253A%252F%252Fid.wpvip.com%252Fwp-json%252Foauth%252Fv1%252Fcallback%252Fwpcom%26client_id%3D76596%26from-calypso%3D1)
[Crowdsignal](http://calypso.localhost:3000/start/crowdsignal/oauth2-name?oauth2_client_id=978&oauth2_redirect=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fclient_id%3D978%26response_type%3Dcode%26blog_id%3D0%26state%3D0eb3f4fa8a636f73b34f4c325da1691565e50548192a021d9614ebfe92669886%26redirect_uri%3Dhttps%253A%252F%252Fapp.crowdsignal.com%252Fconnect%253Fsource%253Dlogin%2526action%253Drequest_access_token%26wpcom_connect%3D1%26from-calypso%3D1)
Partner Portal - Can't link due to GitHub OSS scan rules: go to hosts.automatt_c.com and you should be redirected to log in, then click 'Create an account'. Edit the URL and change the host to calypso.localhost:3000
WP.Cloud - Can't link due to GitHub OSS scan rules: go to [wp.cloud](https://wp.cloud/) and click 'start here', then 'Create an account'. Edit the URL and change the host to calypso.localhost:3000

### Screenshots (showing primary buttons focused)

| Brand | Before | After |
|-|-|-|
| WordPress.com | ![wordpress com_start_account_redirect_to=%2Fsites(5 1440x900)](https://github.com/user-attachments/assets/f09b6138-f2df-4004-a2b2-1168d9e7cb52) | ![calypso localhost_3000_start_account_user-social(5 1440x900)](https://github.com/user-attachments/assets/ca2bde21-6cfc-4c85-8c34-47b32ea19a58) |
| Blaze Pro | ![wordpress com_start_wpcc_oauth2-user_oauth2_client_id=99370 oauth2_redirect=https%3A%2F%2Fpublic-api wordpress com%2Foauth2%2Fauthorize%3Fresponse_type%3Dcode%26client_id%3D99370%26state%3D6e545c7a9932a0a612817faa07992930afae8229e4b11a2 (2)](https://github.com/user-attachments/assets/4f8ddbdd-b855-483f-9ff4-579a65e65fa5) | ![calypso localhost_3000_start_wpcc_oauth2_client_id=99370 oauth2_redirect=https%3A%2F%2Fpublic-api wordpress com%2Foauth2%2Fauthorize%3Fresponse_type%3Dcode%26client_id%3D99370%26state%3D6e545c7a9932a0a612817faa07992930afae8229e4b11a219e9db9](https://github.com/user-attachments/assets/bc9e5b25-3ad6-44f3-83cd-37652f0b2211) |
| Jetpack Cloud | ![wordpress com_start_wpcc_oauth2-user_oauth2_client_id=69040 oauth2_redirect=https%3A%2F%2Fpublic-api wordpress com%2Foauth2%2Fauthorize%3Fresponse_type%3Dtoken%26client_id%3D69040%26redirect_uri%3Dhttps%253A%252F%252Fcloud jetpack com%2 (1)](https://github.com/user-attachments/assets/3615cb7c-5d4d-489c-9643-43d6ec98a9d0) | ![calypso localhost_3000_start_wpcc_oauth2-user_oauth2_client_id=69040 oauth2_redirect=https%3A%2F%2Fpublic-api wordpress com%2Foauth2%2Fauthorize%3Fresponse_type%3Dtoken%26client_id%3D69040%26redirect_uri%3Dhttps%253A%252F%252Fcloud jetpack](https://github.com/user-attachments/assets/940119ed-2c71-4501-b326-533579681290) |
| Akismet | ![wordpress com_start_account_user-social_redirect_to=https%3A%2F%2Fr-login wordpress com%2Fremote-login php%3Faction%3Dlink%26back%3Dhttps%253A%252F%252Fakismet com%252Faccount%252F(5 1440x900) (1)](https://github.com/user-attachments/assets/ef6fa0df-adde-4f28-aab0-de746ee9cbe4) | ![calypso localhost_3000_start_account_user-social_redirect_to=https%3A%2F%2Fr-login wordpress com%2Fremote-login php%3Faction%3Dlink%26back%3Dhttps%253A%252F%252Fakismet com%252Faccount%252F(5 1440x900)](https://github.com/user-attachments/assets/18506b08-6514-4920-9482-4de1471c6924) |
| Woo | ![wordpress com_start_wpcc_oauth2_client_id=50916 oauth2_redirect=https%3A%2F%2Fpublic-api wordpress com%2Foauth2%2Fauthorize%2F%3Fresponse_type%3Dcode%26client_id%3D50916%26state%3D69c7f83c44d9438e013f60308e0d8369725e57fa68d18b558d8b41743976](https://github.com/user-attachments/assets/f11578c0-b5d7-4e9a-81ea-d88e0f6c7537) | ![calypso localhost_3000_start_wpcc_oauth2-user_oauth2_client_id=50916 oauth2_redirect=https%3A%2F%2Fpublic-api wordpress com%2Foauth2%2Fauthorize%2F%3Fresponse_type%3Dcode%26client_id%3D50916%26state%3D36e78dc55c52c1680dc1b15a5e019bddb1baab6](https://github.com/user-attachments/assets/fe411a03-34e0-40bd-96d0-564c723d6c9c) |
| VIP | ![wordpress com_start_wpcc_oauth2-user_oauth2_client_id=76596 oauth2_redirect=https%3A%2F%2Fpublic-api wordpress com%2Foauth2%2Fauthorize%3Fstate%3D2e373ba7d0b99c5b66af232a2cd4a90f3af2aeafbb0326604917c9333ac1%26scope%3Dauth%26response_type%3D](https://github.com/user-attachments/assets/a04d37f4-62fc-46c8-94d8-30177640798d) | ![calypso localhost_3000_start_wpcc_oauth2-user_oauth2_client_id=76596 oauth2_redirect=https%3A%2F%2Fpublic-api wordpress com%2Foauth2%2Fauthorize%3Fstate%3D2e373ba7d0b99c5b66af232a2cd4a90f3af2aeafbb0326604917c9333ac1%26scope%3Dauth%26res (1)](https://github.com/user-attachments/assets/dfed4850-064f-4bfe-878f-9d68268f91c9) |
| A4A | ![wordpress com_start_wpcc_oauth2-user_oauth2_client_id=95931 oauth2_redirect=https%3A%2F%2Fpublic-api wordpress com%2Foauth2%2Fauthorize%3Fresponse_type%3Dtoken%26client_id%3D95931%26redirect_uri%3Dhttps%253A%252F%252Fagencies automattic (1)](https://github.com/user-attachments/assets/0307c639-ae41-465e-8781-81e24def49f5) | ![calypso localhost_3000_start_wpcc_oauth2-user_oauth2_client_id=95931 oauth2_redirect=https%3A%2F%2Fpublic-api wordpress com%2Foauth2%2Fauthorize%3Fresponse_type%3Dtoken%26client_id%3D95931%26redirect_uri%3Dhttps%253A%252F%252Fagencies autom](https://github.com/user-attachments/assets/4140c098-9066-46ee-94ef-0477dec02fdd) |
| WP Cloud | ![wordpress com_start_wpcc_oauth2-user_oauth2_client_id=103914 oauth2_redirect=https%3A%2F%2Fpublic-api wordpress com%2Foauth2%2Fauthorize%3Fresponse_type%3Dtoken%26client_id%3D103914%26redirect_uri%3Dhttps%253A%252F%252Fhostpartnerportal wor](https://github.com/user-attachments/assets/de3ace43-ba7c-4a8f-8de6-d6ce47d18a41) | ![calypso localhost_3000_start_wpcc_oauth2-user_oauth2_client_id=103914 oauth2_redirect=https%3A%2F%2Fpublic-api wordpress com%2Foauth2%2Fauthorize%3Fresponse_type%3Dtoken%26client_id%3D103914%26redirect_uri%3Dhttps%253A%252F%252Fhostpartnerp](https://github.com/user-attachments/assets/95ff4ce5-eeaf-462d-9747-af7091f32e21) |
| Crowdsignal desktop | ![wordpress com_start_crowdsignal_oauth2-name_oauth2_client_id=978 oauth2_redirect=https%3A%2F%2Fpublic-api wordpress com%2Foauth2%2Fauthorize%3Fclient_id%3D978%26response_type%3Dcode%26blog_id%3D0%26state%3D4d83e8088443caf5ec4ff3581e3973 (1)](https://github.com/user-attachments/assets/573c9f80-2620-42c7-b630-38ee4af3bcc2) | ![calypso localhost_3000_start_crowdsignal_oauth2-name_oauth2_client_id=978 oauth2_redirect=https%3A%2F%2Fpublic-api wordpress com%2Foauth2%2Fauthorize%3Fclient_id%3D978%26response_type%3Dcode%26blog_id%3D0%26state%3D0eb3f4fa8a636f73b34f4c325](https://github.com/user-attachments/assets/614dae7f-4dd4-4d88-8b6b-9153d09d7284) |
| Crowdsignal mobile | ![wordpress com_start_crowdsignal_oauth2-name_oauth2_client_id=978 oauth2_redirect=https%3A%2F%2Fpublic-api wordpress com%2Foauth2%2Fauthorize%3Fclient_id%3D978%26response_type%3Dcode%26blog_id%3D0%26state%3D4d83e8088443caf5ec4ff3581e3973 (2)](https://github.com/user-attachments/assets/1c3d9a37-9f61-427e-a89f-dd306e8e0e63) | ![calypso localhost_3000_start_crowdsignal_oauth2-name_oauth2_client_id=978 oauth2_redirect=https%3A%2F%2Fpublic-api wordpress com%2Foauth2%2Fauthorize%3Fclient_id%3D978%26response_type%3Dcode%26blog_id%3D0%26state%3D0eb3f4fa8a636f73b34f4 (1)](https://github.com/user-attachments/assets/782cce80-f656-4739-96f7-d27b933b84e7) |

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

adamwoodnz committed:
Replace signup button with core button

adamwoodnz committed:
Use core button for passwordless signup submit

adamwoodnz committed:
Use core button for crowdsignal signup buttons

adamwoodnz committed:
Stop crowdsignal button style bleed

adamwoodnz committed:
Style Akismet social signup button

adamwoodnz committed:
Add signup submit button stories

adamwoodnz committed:
Use relative imports for sign up form components

adamwoodnz committed:
Reduce scope of logged out footer changes

adamwoodnz committed:
Fix Akismet styles

adamwoodnz committed:
Remove remaining social button overrides

adamwoodnz committed:
Simplify Crowdsignal styles

adamwoodnz committed:
Simplify Akismet styles

adamwoodnz committed:
Scope form button styles so they don't affect core buttons

adamwoodnz commented on client/layout/masterbar/woo.scss:
```diff
@@ -156,7 +156,7 @@ $breakpoint-mobile: 660px;
 	.wp-login__links a,
 	.wp-login__links button,
-	.button:not(.social-buttons__button),
+	.button,
```
These don't exist anymore, all social buttons were migrated to core buttons in https://github.com/Automattic/wp-calypso/pull/100807

adamwoodnz commented on client/signup/style.scss:
```diff
@@ -307,14 +307,14 @@ body.is-section-signup .layout:not(.dops):not(.is-wccom-oauth-flow) {
-		button.signup-form__submit:disabled,
-		button.signup-form__submit.disabled {
+		.button.signup-form__submit[disabled],
+		.button.signup-form__submit:disabled,
+		.button.signup-form__submit.disabled {
 			background-color: var(--studio-gray-20);
 			color: var(--color-surface);
 		}
 
-		button.signup-form__submit {
+		.button.signup-form__submit {
```
Stop these styles bleeding into core buttons. There is still a `.signup-form__submit` to be migrated: https://github.com/Automattic/wp-calypso/blob/3ab6adb2ac69b39cb94b52ad1c55b5ba3f12d7ef/client/my-sites/invites/invite-accept-logged-out/index.jsx#L153

adamwoodnz commented on client/components/logged-out-form/footer.scss:
```diff
@@ -16,4 +18,9 @@
 		@include elevation( 1dp );
 	}
+
+	.components-button.is-primary {
+		width: 100%;
+		justify-content: center;
+	}
```
See discussion: https://github.com/Automattic/wp-calypso/pull/103142#discussion_r2074878723

adamwoodnz committed:
Update brand button copy in stories

adamwoodnz commented on client/blocks/signup-form/style.scss:
```diff
@@ -1,6 +1,7 @@
 @import "@wordpress/base-styles/breakpoints";
 @import "@wordpress/base-styles/mixins";
 @import "@automattic/onboarding/styles/mixins";
+@import "../../assets/stylesheets/shared/mixins/breakpoints";
```
Storybook build of this individual file fails without explicit imports for this dependency

adamwoodnz commented on client/components/logged-out-form/footer.scss:
```diff
@@ -1,3 +1,5 @@
+@import "../../assets/stylesheets/shared/mixins/elevation";
```
Storybook build of this individual file fails without explicit imports for this dependency

adamwoodnz commented on client/jetpack-connect/test/__snapshots__/signup.js.snap:
```diff
@@ -213,8 +213,7 @@ exports[`JetpackSignup should render 1`] = `
               </p>
               <button
-                class="button signup-form__submit form-button is-primary"
-                locale="en"
```
I cannot find any code which depends on this non-standard attribute. I'd expect if it was used it would be data-locale. [It's passed to all FormButton components via the localize higher order component](https://github.com/Automattic/wp-calypso/blob/b159deb4ada78061bb88437f5d7e50e460251c4a/client/components/forms/form-button/index.jsx#L38), and is removed now that we're using a core Button.

adamwoodnz replied:
I cannot find any code which depends on this non-standard attribute. I'd expect if it was used it would be using the standard `data-locale`. [It's passed to all FormButton components via the localize higher order component](https://github.com/Automattic/wp-calypso/blob/b159deb4ada78061bb88437f5d7e50e460251c4a/client/components/forms/form-button/index.jsx#L38), and is no longer added to the DOM now that we're using a core Button. I believe the FormButton should be omitting it when adding props to the element, along with `translate`.

adamwoodnz committed:
Update Jetpack snapshots

adamwoodnz committed:
Fix Blaze link colors

adamwoodnz commented on client/blocks/signup-form/index.jsx:
```diff
@@ -1032,29 +1031,6 @@ class SignupForm extends Component {
-
-		if ( isBlazePro ) {
-			return (
-				<div>
-					<LoggedOutFormLinks>
-						<span>{ this.props.translate( 'Already have an account?' ) }&nbsp;</span>
-						<LoggedOutFormLinkItem href={ this.getLoginLink() }>
-							{ this.props.translate( 'Log in here' ) }
-						</LoggedOutFormLinkItem>
-					</LoggedOutFormLinks>
-				</div>
```
A login link is already rendered above the form, this removes the duplicate.

| Before | After |
|-|-|
| ![wordpress com_start_wpcc_oauth2-user_oauth2_client_id=99370 oauth2_redirect=https%3A%2F%2Fpublic-api wordpress com%2Foauth2%2Fauthorize%3Fresponse_type%3Dcode%26client_id%3D99370%26state%3D6e545c7a9932a0a612817faa07992930afae8229e4b11a2 (2)](https://github.com/user-attachments/assets/4f8ddbdd-b855-483f-9ff4-579a65e65fa5) | ![calypso localhost_3000_start_wpcc_oauth2_client_id=99370 oauth2_redirect=https%3A%2F%2Fpublic-api wordpress com%2Foauth2%2Fauthorize%3Fresponse_type%3Dcode%26client_id%3D99370%26state%3D6e545c7a9932a0a612817faa07992930afae8229e4b11a219e9db9](https://github.com/user-attachments/assets/bc9e5b25-3ad6-44f3-83cd-37652f0b2211) |

These screenshots also show the links colors I fixed @jasmussen, but none of these orange elements, nor the grey terms text meet WCAG contrast requirements:

![Screenshot 2025-05-08 at 4 19 53 PM](https://github.com/user-attachments/assets/a66e840c-9763-462d-9496-56a540662c2b)

jasmussen replied:
Indeed, excellent catch and thank you for surfacing this. This is a great example of _why_ we want to build systems first as much as possible, so we can unify disparate-but-similar interfaces and elevate all of them together.

Regarding the duplicate login link, the end-state design would probably be to use the Calypso screen as a template. That has a log-in link in the top right corner of the entire screen, as persistent affordance across related screens when you're not logged in.

Near term, or small-fix territory, you can either choose to have both of those, or delete the former one. My instinct would be to keep the bottom link in context of the log-in button, not the one near the header. I.e. delete this:

![Screenshot 2025-05-08 at 11 05 13](https://github.com/user-attachments/assets/3fd896ca-7636-481b-bdb0-f582caed21c3)

Keep this:

![Screenshot 2025-05-08 at 11 05 17](https://github.com/user-attachments/assets/1a9645c5-0afd-4b8e-af66-0b850a1c4cab)

In both cases, ensure to end every sentence with a period. I.e. "Already have an account? Log in."

The second issue is **the contrast issue**. I'm not aware of which designer(s) were attached to this in the past that might have additional context. The easiest fix if we are to fix it here, would be blueberry hyperlinks, black text on the orange button (if we are to keep the orange accent color):

![Screenshot 2025-05-08 at 11 04 25](https://github.com/user-attachments/assets/92498751-d144-402b-90a1-2e5ae513e0e7)

Ironically that visually feels like lower contrast though, which just surfaces some of the WCAG algorithms and how they feel dated. In that sense, if the best fix is likely to replace the orange with the blueberry (or a blue from the Blaze palette), and keep the orange brand color for bigger swing contexts. 

Both of those are in redesign territory, though, so the immediate fix may be to do as little as possible for this PR, and then separately file an issue that pings the Blaze team about this issue.

adamwoodnz replied:
Updated in [1c07f9d](https://github.com/Automattic/wp-calypso/pull/103144/commits/1c07f9da8094c4f55cd3bf10adc757370b194e3c)

New screenshot:

![calypso localhost_3000_start_wpcc_oauth2-user_oauth2_client_id=99370 oauth2_redirect=https%3A%2F%2Fpublic-api wordpress com%2Foauth2%2Fauthorize%3Fresponse_type%3Dcode%26client_id%3D99370%26state%3D6e545c7a9932a0a612817faa07992930afae82 (1)](https://github.com/user-attachments/assets/e5d4cc3e-5495-4af3-8101-336905b6a2bd)

jasmussen replied:
Looks great. Can you add a period at the end of the sentence, so it's "First, create your WordPress.com account."?

Not a blocker.

adamwoodnz replied:
Done in [2f08f67](https://github.com/Automattic/wp-calypso/pull/103144/commits/2f08f6721bf59556109ee5ea1675c14e57570f97)

adamwoodnz replied:
Issue created for contrast issues, and added to the audit project: [DOTCOM-13184: Blaze Pro login/signup screens do not meet WCAG 2 AA minimum contrast ratio thresholds](https://linear.app/a8c/issue/DOTCOM-13184/blaze-pro-loginsignup-screens-do-not-meet-wcag-2-aa-minimum-contrast)

chihsuan reviewed (commented):
Thank you for all your work on these improvements. @adamwoodnz 

I've tested the Woo Signup flow, and it looks like the button style of the disabled state has been changed.


Could you take a look? Other than that, everything seems to be working as expected. Thanks.

Before:
<img width="445" alt="Screenshot 2025-05-08 at 14 58 11" src="https://github.com/user-attachments/assets/99788e0b-677e-4989-8307-c5c3556a71ed" />


After:
<img width="499" alt="Screenshot 2025-05-08 at 14 53 55" src="https://github.com/user-attachments/assets/5c581e76-30b8-4029-b7bc-d12bb8886a25" />

adamwoodnz commented:
> I've tested the Woo Signup flow, and it looks like the button style of the disabled state has been changed.
> 
> Could you take a look? Other than that, everything seems to be working as expected. Thanks.

Thanks, I'll fix this in https://github.com/Automattic/wp-calypso/pull/103142 as it's going to ship first

adamwoodnz committed:
Update Blaze login links

- Remove duplicate top link and customise copy
- Fix translation and accessibility of lower link

adamwoodnz commented on client/blocks/signup-form/index.jsx:
```diff
@@ -1041,14 +1040,11 @@ class SignupForm extends Component {
-					</LoggedOutFormLinks>
-				</div>
+				<p className="signup-form__login-link">
+					{ this.props.translate( 'Already have an account? {{link}}Log in here{{/link}}.', {
```
Fixes issues with the translation, as the complete string is now able to be translated.

adamwoodnz committed:
Fix color contrast issue with terms of service copy

adamwoodnz committed:
Remove redundant button content centering

tellthemachines commented on client/jetpack-connect/test/__snapshots__/signup.js.snap:
```diff
@@ -500,8 +499,7 @@ exports[`JetpackSignup should render with locale suggestions 1`] = `
               </p>
               <button
-                class="button signup-form__submit form-button is-primary"
-                locale="en"
```
If this test is meant to test that `JetpackSignup should render with locale suggestions` and we've removed the `locale`, shouldn't we also delete the test?

(I'm not sure this was working correctly to start with as the test sets `locale="es"` and this snapshot had `locale="en"`)

adamwoodnz replied:
Good question. I believe the locale attribute on the button isn't actually used or related to the test. I ran a diff on the snapshot output and the `locale="es"` in the test props results in a different signup URL:

`href="/log-in/jetpack/es?site=http%3A%2F%2Fan.example.site&redirect_to=https%3A%2F%2Fexample.com%2F&email_address=email%40an.example.site&from=banner-44-slide-1-dashboard"`

vs

`href="/log-in/jetpack?site=http%3A%2F%2Fan.example.site&redirect_to=https%3A%2F%2Fexample.com%2F&email_address=email%40an.example.site&from=banner-44-slide-1-dashboard"`

So I think it's still valid.

tellthemachines replied:
Oh that makes sense!

tellthemachines reviewed (approved):
Code LGTM! Just one non-blocking comment on the snapshots below.

Something unrelated to this PR that I noticed: on Crowdsignal and Blaze Pro I'm seeing "Create a site" as the page title. That sounds weird given neither of those products are about creating sites. I'm guessing that's the default message for creating a wordpress.com account but given the diversity of products leveraging this signup we should perhaps make it something more generic like "Create your account"?

adamwoodnz commented:
> Something unrelated to this PR that I noticed: on Crowdsignal and Blaze Pro I'm seeing "Create a site" as the page title. That sounds weird given neither of those products are about creating sites. I'm guessing that's the default message for creating a wordpress.com account but given the diversity of products leveraging this signup we should perhaps make it something more generic like "Create your account"?

Yeah nice spotting, I hadn't noticed that, but I see it on all the sites that use `http://calypso.localhost:3000/start/wpcc/oauth2-user?...` for signup (also Woo, VIP, Jetpack Cloud, etc.)

I'll create an issue.

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/17467615

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @adamwoodnz for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/17467615).
